test/Dialect/FIRRTL/lowerXMR.mlir

```diff
@@ -0,0 +1,67 @@
+// RUN: circt-opt %s  --firrtl-lower-xmr -split-input-file |  FileCheck %s
+
+// Test for same module lowering
+// CHECK-LABEL: firrtl.circuit "xmr"
+firrtl.circuit "xmr" {
+  firrtl.module @xmr(out %o: !firrtl.uint<2>) {
+    %w = firrtl.wire : !firrtl.uint<2>
+    %1 = firrtl.ref.send %w : !firrtl.uint<2>
+    %x = firrtl.ref.resolve %1 : !firrtl.ref<uint<2>>
+    firrtl.strictconnect %o, %x : !firrtl.uint<2>
+    // CHECK:  %w = firrtl.wire sym @xmr_sym   : !firrtl.uint<2>
+    // CHECK-LITERAL:  %0 = firrtl.verbatim.expr "{{0}}" : () -> !firrtl.uint<2> {symbols = [#hw.innerNameRef<@xmr::@xmr_sym>]}
+    // CHECK:  firrtl.strictconnect %o, %0 : !firrtl.uint<2>
+  }
+}
+
+// -----
+
+// Test for multiple readers and multiple instances
```

Comment:
Nit:
For maintainability, it may be better to test these features separately from each other.